test/framework/provider.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"time"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Wait for Provider to be successfully installed.
+func WaitForProviderToBeSuccessfullyInstalled(ctx context.Context, c client.Client) error {
+	if err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		l := &v1.ProviderList{}
+		if err := c.List(ctx, l); err != nil {
+			return false, err
+		}
+		if len(l.Items) != 1 {
+			return false, nil
+		}
+		for _, p := range l.Items {
+			if p.GetCondition(v1.TypeInstalled).Status != corev1.ConditionTrue {
+				return false, nil
+			}
+			if p.GetCondition(v1.TypeHealthy).Status != corev1.ConditionTrue {
+				return false, nil
+			}
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Wait for Provider to be successfully updated.
+func WaitForProviderToBeSuccessfullyUpdated(ctx context.Context, c client.Client, p2 string, p1 string) error {
```

Comment:
Same as above but maybe something like `WaitForRevisionTransition` or something like that?